crypto/dh_extra/dh_test.cc

```diff
@@ -300,33 +301,216 @@ TEST(DHTest, ASN1) {
+          "236eceddf59c7d463071d2715612ed78c35d6e3783da3042862068cf206a08fe"
+          "ce83f60572db55c60f6f6811f359b6f1d504e33d3054c0fe083dc7e73030ed42"
+          "7f079ce60324e71e81fed25c11d2dc853a9fa9f2f64c33f92618d01b8b9bdde6"
+          "2792fbc353aeb97f370f0ef85bbbd0eccbfd7104f6c4e77c7b26ff380aedb4e4"
+          "f974706aa9b4c8eeb924c72d233aa90b8d0376f540ff4af63fa4a7baf47b035f"
+          "f2a1564080e2c31bfc4fad3834d021858e26e4710db3e37144332b909f3340a8"
+          "805b66a7cf042c37797bd46f784793e49cb16e3728e1fc6c98986d21027303ef"
+          "898d32caf0131c323518ced384a9b7ae0c45c15edcb054dfe7044af3ec616ce8"
+          "e5870ea2bef5aa40f4e65721d724ec68774638b13350abbb1f2ac22b0852c6e3"
+          "4ab2608390ec3b971021c0c20e18e2cbcc89b1eea1c2ecb1db6eeaf4195ec7f2"),
+       HexToBytes(
+          "aabbccddeeff11223344556677889900aabbccddeeff11223344556677889900"),
+       HexToBytes(
+          "ba7d5bb3682473327e80c07bcfd58a6af9bf0fa4662288291feb847cc8121ca6"
+          "12ff09bc9d46e3a76f44bad0006e1babdafef5091aed25e53037a9077af93bc5"
+          "76910dbc3e6d345174b36dbec2ab92e0744dc4f5d1d25596b9aa53bc10c22dfe"
+          "fec93c178b8ffd4388c07ffa9ad8a7f22c274066c92f8063b1665609aa224039"
+          "aff15fb5ac07b21f9c81aace529ad5c29688d6940996b5e3a47de1b8cd3b212a"
+          "3e534677df246375679cc014a77c3cc4e14aaa5eb4fc4d0f8a542a0e833a16f4"
+          "dc5c46f11c5ffe14152b9c7f9e504ae01ff84db158b9e48e9fbc46b99190cad2"
+          "e22113797dc7c81ad7c86bcd5e75405226459bb54b26fae179378e377ce5618e"
+          "65f04d2213e1991cefe991ec43272b6c7d93b51e2ccc3bf64486efd1e5c73b3b"
+          "f9344271ab9fbc43af41232ae7524c8213433ef39c64481e9cd06b9f9dc34226"
+          "85dfb2d69b8dc1af0f44d6f52d1d857ec28d93f459a23386ecfe5d97130e201d"
+          "90f159ff5995bbf766ceb38594b7a3192c99432e007b99f1ea7a828d15a1cba6"
+          "d86cd020ff64b1774cd35e33e3696a98574cedb64534f8ca88e2690709718d66"
+          "f4b88d759689819cde545d202b641b0529a02d588ff4c6b832c3f5a3d9bec9ec"
+          "ce0fb9af978b76bf93eba919c5bef844b4b1e2bff3d3758b577c70fa78d89a1d"
+           "d5a1864a2d3795c3668562c67aa77265f38812f001d28b25f7965109481ec2c7")
+      }
+  };
+  for (const testInput &test : testInputs ){
+    bssl::UniquePtr<BIGNUM> client_public(BN_bin2bn(test.client_pk.data(), test.client_pk.size(), nullptr));
+    EXPECT_TRUE(client_public);
+
+    bssl::UniquePtr<BIGNUM> server_secret(BN_bin2bn(test.server_sk.data(), test.server_sk.size(), nullptr));
+    EXPECT_TRUE(server_secret);
 
-  uint8_t buffer[sizeof(kPrime1536)];
-  ASSERT_EQ(BN_num_bytes(bn.get()), sizeof(kPrime1536));
-  ASSERT_EQ(BN_bn2bin(bn.get(), buffer), sizeof(kPrime1536));
-  EXPECT_EQ(Bytes(buffer), Bytes(kPrime1536));
+    bssl::UniquePtr<DH> ffdhe2048_dh(DH_new_by_nid(test.nid));
+    EXPECT_TRUE(DH_set0_key(ffdhe2048_dh.get(), nullptr, server_secret.release()));
+    uint8_t buffer[4096];
+    int size = DH_compute_key(buffer, client_public.get(), ffdhe2048_dh.get());
+    EXPECT_LE(size, 4096);
```

Comment:
`DH_compute_key` returns a negative number on error, so any errors might fall through the gaps here. Although the next test would still fail so..